linalg/ode.hpp

```diff
@@ -12,20 +12,59 @@
 #ifndef MFEM_ODE
 #define MFEM_ODE
 
+#include "../general/communication.hpp"
 #include "../config/config.hpp"
 #include "operator.hpp"
-#include "../general/communication.hpp"
+#include <vector>
+
 
 namespace mfem
 {
 
+/// A class for storing states of previous timesteps
+class StateData
+{
+protected:
+   MemoryType mem_type;
+   int ss, smax;
+   std::vector<Vector> k;
+   Array<int> idx;
+
+public:
+   StateData () { ss = smax = 0;};
+   void  SetSize(int stages, int vsize, MemoryType mem_type);
+   inline void ShiftStages()
+   {
+      for (int i = 0; i < smax; i++) { idx[i] = (++idx[i])%smax; }
+   };
+
+   int  GetMaxSize() { return smax; };
+   int  GetSize() { return ss; };
+   void IncrementSize() { ss++; ss = std::min(ss,smax); };
+   void ResetSize() { ss = 0; };
+
+   const Vector &Get(int i);
+   void Get(int i, Vector &state);
+   void Set(int i, Vector &state);
+   void Add(Vector &state);
+
+   /// Reference access to the ith vector.
+   inline Vector & operator[](int i) { return k[idx[i]]; };
+
+   /// Const reference access to the ith vector.
+   inline const Vector &operator[](int i) const { return k[idx[i]]; };
+
+   void Print(std::ostream &out = mfem::out);
```

Comment:
It would be good to add Doxygen comments for all members (data and functions) here.

Also, it looks like some of these member functions should be `const`.

Reply:
I have added the `const` keywords, where I think they are appropriate.

Will work on the Doxygen, later this week.

Reply:
I have add `///` comments. 
Hope this is sufficient and clear.
Or are `\**` with or wo `brief` preferred?

Reply:
`///` comments are great (both styles are fine), thanks!